packages/nango-yaml/lib/parser.v2.ts

```diff
@@ -0,0 +1,261 @@
+import type {
+    HTTP_VERB,
+    NangoModel,
+    NangoSyncEndpoint,
+    NangoYamlParsedIntegration,
+    NangoYamlV2,
+    NangoYamlV2Integration,
+    ParsedNangoAction,
+    ParsedNangoSync
+} from '@nangohq/types';
+import { NangoYamlParser } from './parser.js';
+import {
+    ParserErrorDuplicateEndpoint,
+    ParserErrorDuplicateModel,
+    ParserErrorEndpointsMismatch,
+    ParserErrorInvalidRuns,
+    ParserErrorModelIsLiteral
+} from './errors.js';
+import { getInterval } from './helpers.js';
+
+export class NangoYamlParserV2 extends NangoYamlParser {
+    parse(): boolean {
+        const yaml = this.raw as unknown as NangoYamlV2;
+        const output: NangoYamlParsedIntegration[] = [];
+        this.modelsParser.parseAll();
```

Comment:
I am gonna say something stupid but since we are basically validating json. Can we not use zod to validate the schema, plus a little bit of post-processing to validate references?

Reply:
I'm sure I get this. Json schema is validating the schema (like zod would do), but all the parsing has almost nothing to do with validation, it's handling typescript syntax inside the yaml